src/algorithms/tonal/audio2pitch.cpp

```diff
@@ -0,0 +1,128 @@
+  _loudnessAlgorithmName = parameter("loudnessAlgorithm").toString();
+  _tolerance = parameter("tolerance").toReal();
+  _pitchConfidenceThreshold = parameter("pitchConfidenceThreshold").toReal();
+  _loudnessThreshold = parameter("loudnessThreshold").toReal();
+  _loudnessThresholdGain = db2amp(_loudnessThreshold);
+
+  if (_maxFrequency > _sampleRate * 0.5) {
+    throw EssentiaException("Audio2Pitch: Max frequency cannot be higher than Nyquist frequency");
+  }
+  if (_maxFrequency <= _minFrequency) {
+    throw EssentiaException("Audio2Pitch: Max frequency cannot be lower or equal than the minimum frequency");
+  }
+
+  if (_pitchAlgorithmName != "pyin_fft" && _pitchAlgorithmName != "pyin") {
+    throw EssentiaException("Audio2Pitch: Bad 'pitchAlgorithm' =", _pitchAlgorithmName);
+  }
+
+  if (_pitchAlgorithmName == "pyin_fft") {
+    _windowing = AlgorithmFactory::create("Windowing");
+    _spectrum = AlgorithmFactory::create("Spectrum");
+    _pitchAlgorithm = AlgorithmFactory::create("PitchYinFFT");
+
+    _windowing->configure("type", "hann",
+                          "size", _frameSize);
+    _spectrum->configure("size", _frameSize);
+  }
+  else {
+    _pitchAlgorithm = AlgorithmFactory::create("PitchYin");
+  }
+
+  if (_loudnessAlgorithmName == "loudness") {
+    _loudnessAlgorithm = AlgorithmFactory::create("Loudness");
+  }
+  else if (_loudnessAlgorithmName == "rms") {
+    _loudnessAlgorithm = AlgorithmFactory::create("RMS");
+  }
+  else {
+    throw EssentiaException("Audio2Pitch: Bad 'loudnessAlgorithm' =", _loudnessAlgorithmName);
+  }
+
+  // switch between pyin and pyin_fft to propagate the weighting parameter
+  if (_pitchAlgorithmName == "pyin") {
+    _pitchAlgorithm->configure(INHERIT("frameSize"),
+                               INHERIT("maxFrequency"),
+                               INHERIT("minFrequency"),
+                               INHERIT("sampleRate"),
+                               INHERIT("tolerance"));
+  }
+  else {
+    _pitchAlgorithm->configure(INHERIT("frameSize"),
+                               INHERIT("maxFrequency"),
+                               INHERIT("minFrequency"),
+                               INHERIT("sampleRate"),
+                               INHERIT("weighting"),
+                               INHERIT("tolerance"));
+  }
+}
+
+void Audio2Pitch::compute() {
+  const std::vector<Real>& frame = _frame.get();
+  Real& pitch = _pitch.get();
+  Real& pitchConfidence = _pitchConfidence.get();
+  Real& loudness = _loudness.get();
+  int& voiced = _voiced.get();
+
+  if (frame.empty()) {
+    throw EssentiaException("Audio2Pitch: cannot compute the pitch of an empty frame");
+  }
+
+  if (frame.size() == 1) {
+    throw EssentiaException("Audio2Pitch: cannot compute the pitch of a frame of size 1");
+  }
+
+  if (_loudnessAlgorithmName == "loudness") {
+    _loudnessAlgorithm->input("signal").set(frame);
+    _loudnessAlgorithm->output("loudness").set(loudness);
+  }
+  else {
```

Comment:
maybe use `else if (_loudnessAlgorithmName == "rms")` to be more explicit (and also it is easier to read the logic)